File: python/examples/hyperliquid.py
from typing import Any

from paysponge import SpongeWallet

from _env import load_dotenv, require_env, sponge_base_url

load_dotenv()

wallet = SpongeWallet.connect(
    api_key=require_env("SPONGE_API_KEY"),
    base_url=sponge_base_url(),
)


def hyperliquid(**kwargs: Any) -> Any:
    if hasattr(wallet, "hyperliquid"):
        return wallet.hyperliquid(**kwargs)
    return wallet._request("POST", "/api/hyperliquid", body=kwargs)


status = hyperliquid(action="status")
markets = hyperliquid(action="markets", limit=5)
positions = hyperliquid(action="positions")
orders = hyperliquid(action="orders")

print("Hyperliquid status")
print(status)

print("Hyperliquid markets")
print(markets)

print("Hyperliquid positions")
print(positions)

print("Hyperliquid open orders")
print(orders)

File: typescript/src/hyperliquid.ts
import { SpongeWallet } from "@paysponge/sdk";
import { requireEnv, spongeBaseUrl } from "./env.js";

const wallet = await SpongeWallet.connect({
  apiKey: requireEnv("SPONGE_API_KEY"),
  baseUrl: spongeBaseUrl(),
});

const status = await wallet.hyperliquid({ action: "status" });
const markets = await wallet.hyperliquid({ action: "markets", limit: 5 });
const positions = await wallet.hyperliquid({ action: "positions" });
const orders = await wallet.hyperliquid({ action: "orders" });

console.log("Hyperliquid status");
console.log(status);

console.log("Hyperliquid markets");
console.log(markets);

console.log("Hyperliquid positions");
console.log(positions);

console.log("Hyperliquid open orders");
console.log(orders);
